wp-includes/class-wp-service-workers.php

```diff
@@ -0,0 +1,212 @@
+<?php
+/**
+ * Dependencies API: WP_Service_Workers class
+ *
+ * @since ?
+ *
+ * @package PWA
+ */
+
+/**
+ * Class used to register service workers.
+ *
+ * @since ?
+ *
+ * @see WP_Dependencies
+ */
+class WP_Service_Workers extends WP_Scripts {
+
+	/**
+	 * Param for service workers.
+	 *
+	 * @var string
+	 */
+	public $query_var = 'wp_service_worker';
+
+	/**
+	 * Output for service worker scope script.
+	 *
+	 * @var string
+	 */
+	public $output = '';
+
+	/**
+	 * WP_Service_Workers constructor.
+	 */
+	public function __construct() {
+		parent::__construct();
+		global $wp_filesystem;
+
+		if ( ! class_exists( 'WP_Filesystem' ) ) {
+			require_once ABSPATH . '/wp-admin/includes/file.php';
+		}
+
+		if ( null === $wp_filesystem ) {
+			WP_Filesystem();
+		}
+	}
+
+	/**
+	 * Initialize the class.
+	 */
+	public function init() {
+		/**
+		 * Fires when the WP_Service_Workers instance is initialized.
+		 *
+		 * @param WP_Service_Workers $this WP_Service_Workers instance (passed by reference).
+		 */
+		do_action_ref_array( 'wp_default_service_workers', array( &$this ) );
+	}
+
+	/**
+	 * Register service worker.
+	 *
+	 * Registers service worker if no item of that name already exists.
+	 *
+	 * @param string $handle   Name of the item. Should be unique.
+	 * @param string $callback Path of the item relative to the WordPress root directory.
```

Comment:
As noted below, this should be:

```
@param string|callable $src URL to the source in the WordPress install, or a callback that returns the JS to include in the service worker.
```

And so it's not actually _relative_. It can be an absolute URL.